### DISCUSSION TITLE
Module Managment

### DISCUSSION THREAD

**Discussion by mucan54** [orchidsoftware/platform#1494] [Ideas]
Maybe some kind of module management can be made for orchid too like this: https://github.com/nWidart/laravel-modules

**Comment by tabuna** [comment]
Hi, what problem do we want to solve?  Can you describe?

**Comment by mucan54** [comment]
No it is not a solution for a problem, you can think it is like a feature, for example I will create a new model with screens and make it as module, and I will be able to use taht module for my future projects or maybe other orchid users can use my modules too.

**Reply by tabuna** [reply]
But then this is no different from the usual Laravel package, only with a dependency on Orchid.

https://github.com/melihovv/laravel-package-generator

On the other hand, if you need a domain-level approach, then this can also be done without using packages.

https://stitcher.io/blog/laravel-beyond-crud-01-domain-oriented-laravel

**Reply by mucan54** [reply]
How I can add new menu items, platform routes, permissions, layouts and screens to existing Orchid project via new Laravel package.

**Reply by tabuna** [reply]
You can heap from this https://github.com/orchidsoftware/platform/blob/58369352a3339d76cfee9e2aadd00f9c71f247c4/src/Platform/OrchidServiceProvider.php#L10 or call the methods of the Dashboard class yourself

**Reply by mucan54** [reply]
How about platform routes?

**Reply by tabuna** [reply]
Just like regular routes.  You can see the crud package example https://github.com/orchidsoftware/crud/blob/efb145f169254e54fbaf4b9ab24b043c7fd6cec7/src/CrudServiceProvider.php#L46

**Reply by bilogic** [reply]
I have to agree that modules are quite pointless:

1. Packages can do everything a module can
2. Developers have to learn another structure on top of Laravel's. The owner of modules doesn't like Laravel structure, but to me, the solution offered is just to create another similar structure that does the same thing
3. Packages can be shared and installed using Composer, I don't see how a module can be shared that way

**Comment by jonatemps** [comment]
how display my permissions in the screen?
